### ISSUE TITLE
Add collectionCode to eventCore

### ISSUE THREAD

**Issue by timrobertson100** [gbif/rs.gbif.org#96]
Originally reported [here](https://github.com/gbif/rs.gbif.org/issues/73#issuecomment-1436950717) by @peterdesmet, providing the justification:

> to indicate the (virtual) collection an event based dataset is derived from

The seems reasonable and fits within the intention from the DwC [collectionCode description](https://dwc.tdwg.org/terms/#dwc:collectionCode):

> The name, acronym, coden, or initialism identifying the collection or data set from which the record was derived.

**Comment by dagendresen** [comment]
Would not dwc:datasetID and dwc:datasetName be more appropriate? Or even better proposing new record-level terms for project and projectID in Darwin Core? Than using collectionCode and collectionID for this purpose?

**Comment by peterdesmet** [comment]
@dagendresen, in my experience `datasetName` and `datasetID` are mainly used for the published dataset itself (title + doi). `project` and `projectID` could be useful, but to indicate the project (cf. project in metadata) not the originating source database.

Indicating the originating source database/system doesn't violate the definition for `collectionCode` in my opinion:

> The name, acronym, coden, or initialism identifying the collection or data set from which the record was derived.

It is already used as such for Occurrence datasets (not only by me, see e.g. `EBIRD` in https://www.gbif.org/occurrence/3504179613). I don't think it makes sense  to exclude it if your dataset happens to be organized as an Event Core dataset, which is why I'm proposing here to include it.

**Comment by dagendresen** [comment]
Would not rather the datasetName and datasetID for the published dataset itself instead belong in the dataset-level EML metadata than as a property for each record?
> datasetName and datasetID are mainly used for the published dataset itself 

Records inside the same dataset could be from different named specimen collections.

**Comment by peterdesmet** [comment]
Yeah, the uses of those fields differ. Note that the IPT does suggest the resource DOI for the `datasetID` (i.e. resource = datasetName/datasetID):

<img width="913" alt="Screenshot 2023-02-22 at 10 59 04" src="https://user-images.githubusercontent.com/600993/220586982-005e3903-0c3c-48b6-b61c-952e0644e1a6.png">

Irrespective on how the fields are actually used, I don't think `collectionCode` and `collectionID` should be excluded if you structure your dataset like an event-core dataset.
